support/controlplane-component/controlplane-component.go

```diff
@@ -253,6 +256,17 @@ func (c *controlPlaneWorkload) update(cpContext ControlPlaneContext) error {
 		return err
 	}
 
+	if c.serviceAccountKubeConfigOpts != nil {
+		_, disablePKIReconciliationAnnotation := cpContext.HCP.Annotations[hyperv1.DisablePKIReconciliationAnnotation]
+		if !disablePKIReconciliationAnnotation {
```

Comment:
I think we need an else here to handle the secret deletion when `hyperv1.DisablePKIReconciliationAnnotation` annotation is added day2

Reply:
I don't think that replicates current behaviour, does it?

Reply:
yeah I don't think we do that in v1, but I think it makes sense to cleanup, no strong feelings though,

Reply:
Let's dont' deviate anything from current behaviour before this is fully migrated and delivered